familias/serializers.py

```diff
@@ -0,0 +1,280 @@
+    """ Serializer to represent a .models.Familia instance
+        through a REST endpoint for the offline application
+        to submit information.
+    """
+    integrante_familia = IntegranteSerializer(many=True, allow_null=True)
+    comentario_familia = ComentarioSerializer(many=True, allow_null=True)
+
+    class Meta:
+        model = Familia
+        fields = (
+            'id',
+            'numero_hijos_diferentes_papas',
+            'explicacion_solvencia',
+            'estado_civil',
+            'localidad',
+            'comentario_familia',
+            'integrante_familia',)
+
+        extra_kwargs = {"id": {"read_only": False, "required": False}}
+
+    def create(self):
+        """ This function overides the default behaviour for creating
+            an object through a serializer.
+
+            Django Rest Framework does not implement saving nested
+            serializers.
+
+            Creates a Family instance. There are objects related to
+            the family that depend upon the creation of a family
+            instance first. This are the Integrantes and Comentarios.
+
+            We first remove all Integrante and comentario from the
+            validated data. We create the family and then we create
+            this objects that require the family instance.
+        """
+        integrantes = self.validated_data.pop('integrante_familia')
+        comentarios = self.validated_data.pop('comentario_familia')
+
+        family_instance = Familia.objects.create(**self.validated_data)
+
+        save_foreign_relationship(integrantes, IntegranteSerializer, Integrante, family_instance)
+        save_foreign_relationship(comentarios, ComentarioSerializer, Comentario, family_instance)
+
+        return family_instance
+
+    def update(self):
+        """ This function overides the default behaviour for creating
+            an object through a serializer.
+
+            Django Rest Framework does not implement saving nested
+            serializers.
+
+            Updates a familia object and all other objects that
+            depend on it. Since the offline client will submit a
+            complete JSON of the study each time.
+
+            When an offline client submits an update of a study,
+            new information can be created. save_foreign_relationship
+            looks for the id in the object. If there is no data it will
+            create the object.
+
+            Integrantes has a non-destructive way of disactivating. This
+            is donde by changin the is_active field.
+
+            Comentario does not have this field, so all comentario instances
+            that were not sent by the offline application most be removed.
+        """
+        integrantes = self.validated_data.pop('integrante_familia')
+        comentarios = self.validated_data.pop('comentario_familia')
+
+        Comentario.objects.exclude(id__in=[comment.get('id') for comment in comentarios]).delete()
```

Comment:
you have to make sure all the selected comentarios have the current familia associate right? otherwise this deletes all comments not necessarily related to this family.